docs/tutorial/resift-rentals.md

```diff
@@ -0,0 +1,2110 @@
+
+```js
+// App.js
+...
+import { useEffect } from 'react';
+import { useDispatch } from 'resift'
+
+function App() {
+  ...
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getGenres());
+  }, [dispatch]);
+}
+```
+
+The complete code at this point looks like this:
+
+```js
+// App.js
+import React, { useEffect } from 'react';
+// Components
+import AppBar from 'components/AppBar';
+import Genre from 'components/Genre';
+// Fetches
+import { useData, useDispatch, Guard } from 'resift';
+import getGenres from 'fetches/getGenres';
+// Styles
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(theme => ({
+  root: {},
+  genre: {
+    margin: '24px 0',
+  },
+  spinner: {
+    color: 'white',
+  },
+}));
+
+function App() {
+  const classes = useStyles();
+  const genres = useData(getGenres);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getGenres());
+  }, [dispatch]);
+
+  return (
+    <div className={classes.root}>
+      <AppBar />
+      <Guard fetch={getGenres}>
+        {genres => genres.map(genre => <Genre key={genre.id} genre={genre} />)}
+      </Guard>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Refresh the page now and you'll see the genres data load after a second. Wait, that's not a great user experience having to wait for the data to come without know what's happening. We should indicate to our user that the data is loading.
+
+![Waiting GIF](https://media.giphy.com/media/9SIXFu7bIUYHhFc19G/giphy.gif)
+
+#### Show Fetch Status
+
+We want to show a loading spinner when we're fetching the data. To achieve this, we'll use the `useStatus` and a helper function to check is the status is 'loading' called `isLoading`. And we'll grab the Material UI spinner called `CircularProgress`. Let's get them imported.
+
+```js
+import { useStatus, isLoading } from 'resift';
+import { CircularProgress } from '@material-ui/core';
+```
+
+Now we can add the spinner in while loading
+
+```js
+function App() {
+  ...
+  const status = useStatus(getGenres)
+
+  return (
+    <>
+      ...
+      {isLoading(status) && <CircularProgress className={classes.spinner} />}
+      ...
+    </>
+  );
+}
+```
+
+Our complete file looks like this now:
+
+```js
+// App.js
+import React, { useEffect } from 'react';
+// Components
+import AppBar from 'components/AppBar';
+import Genre from 'components/Genre';
+// Fetches
+import { useData, useDispatch, Guard, useStatus, isLoading } from 'resift';
+import getGenres from 'fetches/getGenres';
+// Styles
+import { makeStyles } from '@material-ui/core/styles';
+import { CircularProgress } from '@material-ui/core';
```

Comment:
`CircularProgress` is a component and you have it under the `// Styles` comment. Was this intended?

Reply:
I grouped to under //styles because it's from the material ui library